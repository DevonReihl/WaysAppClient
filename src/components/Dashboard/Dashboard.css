.upperSection {
  display: flex;
  flex-direction: column;
  align-items: center;
  padding-top: 20px;
}

.tripSearchBar {
  font-size: 20px;
  font-weight: bold;
  margin-top: 25px;
  margin-bottom: 25px;
  box-sizing: border-box;
}

.lowerSection {
  background-color: rgb(255, 255, 255);
  display: flex;
  flex-direction: column;
  padding: 22px;
  width: 100%;
}

.myButton {
  font-size: 19px;
  letter-spacing: 2px;
  text-transform: uppercase;
  display: inline-block;
  text-align: center;
  width: 180px;
  margin-left: 65px;
  margin-right: 65px;
  font-weight: bold;
  padding: 12px 0px;
  color: rgb(230, 31, 130);
  border: 2px solid rgb(230, 31, 130);
  border-radius: 2px;
  position: relative;
}

.myButton:before {
  -webkit-transition: 0.5s all ease;
  transition: 0.5s all ease;
  position: absolute;
  top: 0;
  left: 50%;
  right: 50%;
  bottom: 0;
  opacity: 0;
  content: '';
  background-color: rgb(230, 31, 130);
  z-index: -2;
}
.myButton:hover:before {
  -webkit-transition: 0.5s all ease;
  transition: 0.5s all ease;
  left: 0;
  border-style: none;
  right: 0;
  opacity: 1;
}
.myButton:focus:before {
  transition: 0.5s all ease;
  left: 0;
  right: 0;
  opacity: 1;
}

.myButton:hover {
  color: #fff;
}

.myButton:focus {
  color: #fff;
}

div.lowerSection {
  min-height: 80vh;
}

div.bttn {
  color: rgb(230, 31, 130);
  text-decoration: none;
  -webkit-transition: 0.3s all ease;
  transition: 0.3s ease all;
}

div.bttn:hover {
  color: #fff;
}

div.bttn:focus {
  color: #fff;
}

.bttn {
  font-size: 19px;
  letter-spacing: 2px;
  text-transform: uppercase;
  display: inline-block;
  text-align: center;
  width: 180px;
  margin-left: 65px;
  margin-right: 65px;
  font-weight: bold;
  padding: 12px 0px;
  color: rgb(230, 31, 130);
  border: 2px solid rgb(230, 31, 130);
  border-radius: 2px;
  position: relative;
  box-shadow: 0px 2px 4px rgb(195, 195, 195);
}

.bttn:before {
  -webkit-transition: 0.5s all ease;
  transition: 0.5s all ease;
  position: absolute;
  top: 0;
  left: 50%;
  right: 50%;
  bottom: 0;
  opacity: 0;
  content: '';
  background-color: rgb(230, 31, 130);
  z-index: -2;
}
.bttn:hover:before {
  -webkit-transition: 0.5s all ease;
  transition: 0.5s all ease;
  left: 0;
  border-style: none;
  right: 0;
  opacity: 1;
}

.bttn:focus:before {
  transition: 0.5s all ease;
  left: 0;
  right: 0;
  opacity: 1;
}

@media all and (min-width: 800px) {
  .lowerSection {
    display: flex;
    flex-direction: row;
    justify-content: space-evenly;
    flex-wrap: wrap;
  }
  .upperSection {
    display: flex;
    flex-direction: row;
    align-items: flex-end;
    justify-content: center;
  }
  .tripSearchBar {
    margin-top: 0;
    margin-bottom: 0;
  }
}

@media all and (max-width: 400px) {
  .myButton {
    font-size: 16px;
    width: 160px;
    padding: 8px 0px;
  }
  .lowerSection {
    padding: 0px;
  }
}
