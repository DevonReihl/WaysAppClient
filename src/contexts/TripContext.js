import React from 'react';
import UserContext from './UserContext';

const TripContext = React.createContext({
  trips: [],
  currTripId: null,
  setCurrTripId: () => {},
  setTrips: () => {},
  verifyAuth: () => {},
  setLoading: () => {},
  returnUserID: () => {},
  setTripRating: () => {},
  loading: null,
});

export default TripContext;

export class TripProvider extends React.Component {
  state = {
    trips: [],
    currTripId: null,
    currPlace: {},
    stops: [],
    loading: false,
  };

  static contextType = UserContext;

  setTrips = (res) => {
    this.setState({
      trips: res,
    });
  };

  setTripRating = (trip_id) => {
    const index = this.state.trips.findIndex((trip) => {
      return trip.id === Number(trip_id);
    });
    let trips = [...this.state.trips];
    trips[index] = {
      ...trips[index],
      rating: Number(trips[index].rating) + 1,
    };
    this.setState({ trips });
  };

  setLoading = (boolean) => {
    this.setState({
      loading: boolean,
    });
  };

  setCurrTripId = (id) => {
    this.setState({
      currTripId: id,
    });
  };

  handleAddTrip = (trip) => {
    this.setState({
      trips: [...this.state.trips, trip],
    });
  };

  verifyAuth = (id) => {
    return id === this.context.user.id;
  };

  returnUserID = () => {
    return this.context.user.id;
  };

  render() {
    const value = {
      trips: this.state.trips,
      currTripId: this.state.currTripId,
      setTrips: this.setTrips,
      setCurrTripId: this.setCurrTripId,
      verifyAuth: this.verifyAuth,
      loading: this.state.loading,
      setLoading: this.setLoading,
      returnUserID: this.returnUserID,
      setTripRating: this.setTripRating,
    };

    return (
      <TripContext.Provider value={value}>
        {this.props.children}
      </TripContext.Provider>
    );
  }
}
