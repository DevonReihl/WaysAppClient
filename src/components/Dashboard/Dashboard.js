import React from 'react';
import { Link } from 'react-router-dom';
import TripContext from '../../contexts/TripContext';
import TripCards from '../TripCards/TripCards';
import './Dashboard.css';

export default class Dashboard extends React.Component {
  constructor() {
    super();
    this.state = {
      error: null,
      searchQuery: '',
      filteredTrips: [],
      resultsFound: true,
    };
  }

  static contextType = TripContext;

  setQuery = (e) => {
    let value = e.target.value;
    this.setState(
      {
        filteredTrips: this.context.trips.filter((trip) => {
          return (
            trip.destination.toLowerCase().includes(value.toLowerCase()) ||
            trip.short_description
              .toLowerCase()
              .includes(value.toLowerCase()) ||
            trip.activities.toLowerCase().includes(value.toLowerCase())
          );
        }),
      },
      () => {
        // if else statement to determine if search results are populated
        if (this.state.filteredTrips.length > 0)
          this.setState({ resultsFound: true });
        else {
          this.setState({ resultsFound: false });
        }
      }
    );
  };

  render() {
    let tripsToMap = [];
    if (this.state.filteredTrips.length > 0) {
      tripsToMap = this.state.filteredTrips;
    } else {
      tripsToMap = this.context.trips;
    }
    let tripCards = tripsToMap.map((trip, index) => {
      return (
        <TripCards
          key={trip.id}
          id={trip.id}
          index={index}
          days={trip.days}
          rating={trip.rating}
          destination={trip.destination}
          activities={trip.activities}
          short_description={trip.short_description}
          image={trip.img}
        />
      );
    });
    return (
      <section className="Dashboard">
        <div className="upperSection">
          <div className="addTripButton">
            <Link to="/add-trip">
              <div className="flex">
                <div className="bttn">Add Trip</div>
              </div>
            </Link>
          </div>
          <div className="tripSearchBar">
            <label htmlFor="tripSearchBar">Search Bar</label>
            <input
              type="text"
              placeholder={'Search for a destination...'}
              name="tripSearchBar"
              id="tripSearchBar"
              onChange={(e) => {
                this.setQuery(e);
              }}
            ></input>
          </div>
          <div className="myTripButton">
            <Link to="/my-trips">
              <div className="flex">
                <div className="bttn">My Trips</div>
              </div>
            </Link>
          </div>
        </div>
        <div className="lowerSection">
          {this.state.resultsFound ? tripCards : 'no results! search again!'}
        </div>
      </section>
    );
  }
}
