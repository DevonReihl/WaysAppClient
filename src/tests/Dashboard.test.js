import React from 'react'
import Dashboard from '../components/Dashboard/Dashboard'
import { shallow } from 'enzyme'
import TripContext from '../contexts/TripContext'
import toJson from 'enzyme-to-json'

describe(`Dashboard component`, () => {
    it('Renders without crashing', () => {
        shallow(
            <TripContext.Provider>
                <Dashboard />
            </TripContext.Provider>
        )
    })
    it('Matches snapshot', () => {
        const wrapper =  shallow(
        <TripContext.Provider>
        <Dashboard />
        </TripContext.Provider>
        )
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})