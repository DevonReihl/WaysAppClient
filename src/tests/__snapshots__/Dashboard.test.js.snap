// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component Matches snapshot 1`] = `<Dashboard />`;
